pkg/controller/nodenetworkconfigurationpolicy/policyconditions/conditions.go

```diff
@@ -93,31 +93,33 @@ func Update(cli client.Client, policy *nmstatev1alpha1.NodeNetworkConfigurationP
 		}
 
 		nodes := corev1.NodeList{}
-		nodeSelectorFilter := client.MatchingLabels(policy.Spec.NodeSelector)
-		err = cli.List(context.TODO(), &nodes, nodeSelectorFilter)
+		err = cli.List(context.TODO(), &nodes)
 		if err != nil {
 			return errors.Wrap(err, "getting nodes failed")
 		}
-		numberOfNodesAffected := len(nodes.Items)
+
+		numberOfNodes := len(nodes.Items)
+		numberOfEnactments := len(enactments.Items)
 
 		// Let's get conditions with true status count
 		enactmentsCount := enactmentconditions.Count(enactments)
 
-		logger.Info(fmt.Sprintf("enactments count: %s", enactmentsCount.String()))
-
-		if numberOfNodesAffected == 0 {
-			message := "Policy does not match any node"
-			setPolicyNotMatching(&policy.Status.Conditions, message)
-		} else if enactmentsCount.Failed()+enactmentsCount.Available() == numberOfNodesAffected {
-			if enactmentsCount.Failed() > 0 {
-				message := fmt.Sprintf("%d/%d nodes failed to configure", enactmentsCount.Failed(), numberOfNodesAffected)
+		logger.Info(fmt.Sprintf("enactments count: %s", enactmentsCount))
+		if numberOfEnactments < numberOfNodes {
```

Comment:
We should not forget to add the NNCE cleanup to Node controller. Otherwise, when someone removes a node, we would return here too quick.

Also, using only the number of NNCEs makes me unsure. Can we just check if `count.Available() + count.Degraded() + count.NotMatching() == numberOfNodes` here? Then you can get rid of the last `else` and move the nested `if else` block one layer higher.

Reply:
I like it we get the progressing at only one place, although is `numberOfFinished < numberOfNodes`

Reply:
Ah, you are right of course :)